[File System Access article] Native File System → File System Access

tomayac opened the pull request:
Changes proposed in this pull request:

- Native File System → File System Access

tomayac commented on src/site/content/en/blog/file-system-access/index.md:
```diff
@@ -19,46 +26,49 @@ feedback:
-API allows web apps to read or save changes directly to files and folders
-on the user's device. Beyond reading and writing files, the File System Access
-API provides the ability to open a directory and enumerate its contents.
+The [File System Access API][spec] (formerly known as the Writeable Files API) enables developers to
```
```suggestion
The [File System Access API][spec] (formerly known as Native File System API
and prior to that Writeable Files API) enables developers to
```

jpmedley commented:
@petele CAn you sign off on this before I look at it?

tomayac commented:
> @petele CAn you sign off on this before I look at it?

@jpmedley, since Pete is OoO for the next couple days, mind having a look now? People have [started complaining](https://github.com/GoogleChrome/web.dev/pull/3906#pullrequestreview-492156023) about the outdated content.

jpmedley commented:
Sorry for the delay. I just now got back to this. @petele is back in the office. Maybe he can get to it now.

jpmedley commented on src/site/content/en/blog/file-system-access/index.md:
```diff
@@ -210,83 +215,147 @@ async function writeURLToFile(fileHandle, url) {
+
+## Polyfilling
+
+It is not possible to completely polyfill the File System Access API methods.
```
This implies that I can polyfill its properties. Is that true?

tomayac replied:
Not the properties, but the behavior. Mostly. I've outlined in the bullets what's possible and what not.

jpmedley reviewed (changes requested):
It looks good. I had a few nits and one that's not a nit: It looks like this is only available to desktop in 86. I don't see any mention of that.

tomayac commented on src/site/content/en/blog/file-system-access/index.md:
```diff
@@ -20,45 +27,49 @@ feedback:
-  System Access API to ensure that people can easily manage their files. See the
-  [security and permissions](#security-considerations) section
-  for more information.
+and prior to that Writeable Files API) enables developers to
```
The API was initially called Writeable Files API, then Native File System API, and finally now we settled on File System Access API.

tomayac commented on src/site/content/en/blog/file-system-access/index.md:
```diff
@@ -210,83 +215,147 @@ async function writeURLToFile(fileHandle, url) {
+### Deleting files and folders in a directory
+
+If you have obtained access to a directory, you can delete the contained files and folders with the
+[`removeEntry()`][removeentry] method, for folders optionally recursively including all subfolders
+and the therein contained files.
```
```suggestion
[`removeEntry()`][removeentry] method. For folders, deletion can optionally be recursive
and include all subfolders and the therein contained files.
```

tomayac commented on src/site/content/en/blog/file-system-access/index.md:
```diff
@@ -210,83 +215,147 @@ async function writeURLToFile(fileHandle, url) {
+The origin-private file system is a storage endpoint that, as the name suggests, is private to the
+origin of the page. While browsers will typically implement this by persisting the contents of this
+origin-private file system to disk somewhere, it is _not_ intended that the contents be easily user
+accessible. Similarly, there is _no_ expectation that files or directories with names matching the
+names of children of the origin-private file system exist. You can operate as usual on the
```
What this is saying is that while the browser might make it *seem* that there are files, internally—since this is an origin-private file system—the browser might store these "files" in a database or any other data structure. Essentially: what you create with this API, don't expect to find it 1:1 somewhere on the hard disk.

jpmedley replied:
Then please say it the way you just said it me and we'll be done.

tomayac commented:
@jpmedley Thanks for the review, I've answered your questions and addressed your remarks. It'd be great if we get this out soon, since the API has launched and the currently live article is outdated.

tomayac commented on src/site/content/en/blog/file-system-access/index.md:
```diff
@@ -210,83 +215,147 @@ async function writeURLToFile(fileHandle, url) {
+The origin-private file system is a storage endpoint that, as the name suggests, is private to the
+origin of the page. While browsers will typically implement this by persisting the contents of this
+origin-private file system to disk somewhere, it is _not_ intended that the contents be easily user
+accessible. Similarly, there is _no_ expectation that files or directories with names matching the
+names of children of the origin-private file system exist. You can operate as usual on the
```
```suggestion
accessible. Similarly, there is _no_ expectation that files or directories with names matching the
names of children of the origin-private file system exist.
While the browser might make it seem that there are files, internally—since this is an origin-private file system—the browser might store these "files" in a database or any other data structure.
Essentially: what you create with this API, do *not* expect to find it 1:1 somewhere on the hard disk.
You can operate as usual on the
```

tomayac commented on src/site/content/en/blog/file-system-access/index.md:
```diff
@@ -210,83 +215,150 @@ async function writeURLToFile(fileHandle, url) {
+origin-private file system to disk somewhere, it is _not_ intended that the contents be easily user
+accessible. Similarly, there is _no_ expectation that files or directories with names matching the
+names of children of the origin-private file system exist.
+While the browser might make it seem that there are files, internally—since this is an origin-private file system—the browser might store these "files" in a database or any other data structure.
```
```suggestion
While the browser might make it seem that there are files, internally—since this is an
origin-private file system—the browser might store these "files" in a database or any
other data structure.
```

tomayac commented on src/site/content/en/blog/file-system-access/index.md:
```diff
@@ -20,45 +27,49 @@ feedback:
-  System Access API to ensure that people can easily manage their files. See the
-  [security and permissions](#security-considerations) section
-  for more information.
+and prior to that Writeable Files API) enables developers to
```
```suggestion
and prior to that it was called Writeable Files API) enables developers to
```

tomayac commented on src/site/content/en/blog/file-system-access/index.md:
```diff
@@ -1,11 +1,18 @@
+  folders on the user's device.
 date: 2019-08-20
-updated: 2020-09-02
+updated: 2020-09-28
```
```suggestion
updated: 2020-10-02
```

tomayac commented:
Thank you, @jpmedley, for the review! I'm merging this as soon as tests have run.
